source/fab/steps/compile_c.py

```diff
@@ -56,7 +57,19 @@ def __init__(self, compiler: Optional[str] = None, common_flags: Optional[List[s
         """
         super().__init__(name=name)
 
-        self.compiler = compiler or os.getenv('CC', 'gcc -c')
+        self.compiler, compiler_flags = get_tool(compiler or os.getenv('CC', 'gcc -c'))  # type: ignore
```

Comment:
Should there be a `get_c_compiler` to mirror the `get_fortran_compiler`? Or should the latter be replaced by `get_tool`?

Reply:
We could definitely remove `get_fortran_compiler` and just call `get_tool(compiler or os.getenv('FC', ''))`. In fact this PR did that at one point. The reason I added `get_fortran_compiler` was as a convenience function for user configs. I think it improves ease and readability in user code. Happy to remove if preferred, of course.

Reply:
At some point people will start to use C++ modules. That will precipitate exactly the same issues surrounding Fortran modules. We will then need per-compiler builds. In the meantime we can leave it as is.